crates/runtime/src/accelerated_table/refresh_task_runner.rs

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use super::refresh_task::RefreshTask;
+use futures::future::BoxFuture;
+use tokio::{
+    select,
+    sync::mpsc::{self, Receiver, Sender},
+    task::JoinHandle,
+};
+
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+use datafusion::{datasource::TableProvider, sql::TableReference};
+
+use super::refresh::Refresh;
+
+pub struct RefreshTaskRunner {
+    dataset_name: TableReference,
+    federated: Arc<dyn TableProvider>,
+    refresh: Arc<RwLock<Refresh>>,
+    accelerator: Arc<dyn TableProvider>,
+    task: Option<JoinHandle<()>>,
+}
+
+impl RefreshTaskRunner {
+    #[must_use]
+    pub fn new(
+        dataset_name: TableReference,
+        federated: Arc<dyn TableProvider>,
+        refresh: Arc<RwLock<Refresh>>,
+        accelerator: Arc<dyn TableProvider>,
+    ) -> Self {
+        Self {
+            dataset_name,
+            federated,
+            refresh,
+            accelerator,
+            task: None,
+        }
+    }
+
+    pub fn start(&mut self) -> (Sender<()>, Receiver<super::Result<()>>) {
```

Comment:
Let's add an `assert!()` that `self.task` is `None`

Reply:
Confirm I'll add this as part of next PR - thanks!